src/common/date-range/DateRangePicker.js

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import DayPicker, { DateUtils } from 'react-day-picker';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import 'react-day-picker/lib/style.css';
+
+import injectT from '../../../app/i18n/injectT';
+import {
+  validateRange,
+  DatePickerValidationError,
+  isValidRange,
+} from './utils';
+import { createNotification } from '../notification/utils';
+import { NOTIFICATION_TYPE } from '../notification/constants';
+
+const DateRangePicker = ({
+  fullDay,
+  minDays,
+  maxDays,
+  startingTime,
+  endingTime,
+  startingDays,
+  openingPeriods,
+  reservations,
+  onRangeSelect,
+  renderChildComponent,
+  initialMonth,
+  t,
+  locale,
+}) => {
+  const [selectedDateRange, setSelectedDateRange] = useState({
+    from: undefined,
+    to: undefined,
+  });
+
+  const handleDayClick = (day, modifiers = {}) => {
+    // Day not available: don't permit selection.
+    if (!modifiers.available) {
+      return;
+    }
+
+    // Not a starting day and no starting day selected yet: make no selection.
+    if (!modifiers.startingDay && !selectedDateRange.from) {
+      return;
+    }
+
+    // Starting day selected, but attempting to select day before the starting day: don't update selection
+    if (
+      selectedDateRange.from
+      && !selectedDateRange.to
+      && day < selectedDateRange.from
+    ) {
+      return;
+    }
+
+    // Both starting day and ending day selected: reset selection to empty.
+    if (selectedDateRange.from && selectedDateRange.to) {
+      setSelectedDateRange({ from: undefined, to: undefined });
+      return;
+    }
+
+    const range = DateUtils.addDayToRange(day, selectedDateRange);
+    range.from && range.from.setHours(startingTime.hours, startingTime.min);
+    range.to && range.to.setHours(endingTime.hours, endingTime.min);
+    setSelectedDateRange(range);
+
+    if (isValidRange(range)) {
+      const validationErrors = validateRange(range, {
```

Comment:
The function name `isValidRange` confused me a little bit especially that we called `validateRange` after it.
Would `isRange` be more descriptive?

Reply:
Changed to `isRange`.